hydrolib/core/io/polyfile/models.py

```diff
@@ -76,6 +76,35 @@ class PolyObject(BaseModel):
     metadata: Metadata
     points: List[Point]
 
+    @property
+    def xy_coordinates(self) -> Tuple[List[float], List[float]]:
+        """Get a tuple containing the x-coordinates and y-coordinates of the polygon points.
+
+        Returns:
+            Tuple[List[float], List[float]]: Tuple containing a list of x-coordinates and a list of y-coordinates
+        """
+
+        x, y, z = self.xyz_coordinates
+        return x, y
```

Comment:
Maybe it would be nice to return it as a dict (or a pandas dataframe): {'x':x,'y':y}
In this case the user also knows what is inside. But maybe that would be unproper programming?

Reply:
I think it is indeed not very clean to return a dictionary. We want to "shield" users from using strings to retrieve data, since that is error prone and requires that the user has knowledge about the key words. 

In this case, I would prefer returning a custom object over a dictionary such as:

```
class XYZValues:
    x: List[float] = []
    y: List[float] = []
    z: List[Optional[float]] = []

    def append(self, x: float, y: float, z: Optional[float]):
        self.x.append(x)
        self.y.append(y)
        self.z.append(z)
```

Which can be used as:
```
values = poly_object.xyz_coordinates
ax.plot(values.x, values.y)
```

As opposed to using a dict:
```
values = poly_object.xyz_coordinates
ax.plot(values["x"], values["y"])
```

Please let me know what you think.

Reply:
Also, if I try out your code, this does not fail:
```
xyz = XYZValues()
xyz.append(x=[1,2],y='a',z=[None,1])
```

Reply:
I can also remove the append function on the XYZValues object and just set up the object correctly internally:

```
        values = XYZValues()

        for point in self.points:
            values.x.append(point.x)
            values.y.append(point.y)
            values.z.append(point.z)

        return values
```

I removed it in the last commit:
https://github.com/Deltares/HYDROLIB-core/commit/114cf2cd0f3434f9accabc249c6bc4afac8ff4f0